Core/tests/Unit/RequestWrapperTest.php

```diff
@@ -668,6 +669,81 @@ public function testPassingInRetryOptions()
         $this->assertTrue($retryFunctionCalled);
         $this->assertTrue($retryListenerCalled);
     }
+
+    /**
+     * Test to verify auth related headers fetching and updation if
+     * credential fetcher implements UpdateMetadataInterface too, else
+     * going ahead with access_code fetching followed by auth header
+     * creation and updation.
+     */
+    public function testFetchingCredentialAsAuthHeader()
+    {
+        $credentialsFetcher = $this->prophesize(FetchAuthTokenInterface::class);
+        $credentialsFetcher->willImplement(UpdateMetadataInterface::class);
+
+        $accessToken = 'abc';
+        $credentialsFetcher->updateMetadata(Argument::cetera())->willReturn([
+            'authorization' => ['Bearer ' . $accessToken]
+        ]);
```

Comment:
We should also add a `shouldBeCalled` on here
```suggestion
        ])->shouldBeCalledOnce();
```

Reply:
Makes sense, fixed.